### PR TITLE
Add ability to disable Left Click to bank inventory, equipment, and looting bag options

### LINKED ISSUE #2867: Option to disable left-click deposit inventory / deposit equipment
Title. Left-click deposit inventory / equipment buttons would only work when chosen from a right-click menu, like Konduit has it.

### PR DESCRIPTION
**Pull request by Hydrox6**
Closes #2867

When the options are enabled, the relevant Deposit options are set to open the right click menu, like the Bank Fillers currently do.

In Action:

This is in it's own plugin due to not really being a Menu Entry swap, and I'm tempted to merge the other bank plugins into it (Bank Value, Bank Tags), though I'm not sure if merging Bank Tags would be a good idea. Open to ideas on that front.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	@Provides
+	BankConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(BankConfig.class);
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if ((event.getOption().equals("Deposit worn items") && config.rightClickBankEquip())
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Make a constant for these strings

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import net.runelite.client.plugins.PluginDescriptor;
+
+
+@PluginDescriptor(
+	name = "Bank Modifications",
+	enabledByDefault = false
+)
+public class BankPlugin extends Plugin
+{
+
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: No newline needed here

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.bank;
+
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: No newline needed here

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import net.runelite.api.Client;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: No newline needed here

**Comment by TzZek** [comment]
I personally think it'd be best to have all options that involve the bank in general in one tab. Rather than having to go to a few different configs for bank related things.

**Comment by Hydrox6** [comment]
@TzZek yeah, I mentioned that at the bottom of the PR message. I'll at least merge Bank Value if this plugin is merged (unless an admin tells me otherwise). Not sure about Bank Tags, because merging it could break current tags if not handled correctly.

**Comment by jpabeem** [comment]
Love this, nice work!

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java (hunk 3 above)
Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Whoops, don't know how I missed this one

**Comment by jplsek** [comment]
I'm not sure if there is currently a clean way for synced configs to migrate to a new key configuration, so it may be difficult to merge the plugins.

There are also plugins that somewhat affects the bank, like "Item Prices", but wouldn't necessarily be part of an overall "Bank" plugin.

There is also a [PR](https://github.com/runelite/runelite/pull/2444) to create plugin categories, and if it gets merged, there could be a "Bank" category that could list all these plugins.

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ */
+package net.runelite.client.plugins.bank;
+
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup(
+	keyName = "bankMods",
+	name = "Bank Modifications",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Maybe merging this with Bank Value plugin would make sense? Its bank modification too after all.

**Comment by Hydrox6** [comment]
Accidentally forgot about the `Bank loot` button in the Deposit Boxes. Added with my last commit.

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java (hunk 5 above)
Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Just did that with my latest commit.

**Comment by Hydrox6** [comment]
Updated and rebased this PR, removed the merging of Bank Value as it would probably be better in its own PR.

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+@ConfigGroup("bankMods")
+public interface BankConfig extends Config
+{
+	@ConfigItem(
+		keyName = "rightClickBankInventory",
+		name = "Disable left click bank inventory",
+		description = "Configures whether the bank inventory button will bank your inventory on left click",
+		position = 0
+	)
+	default boolean rightClickBankInventory()
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: These should all just be an enum with default choice if this even approved

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+		return false;
+	}
+
+	@ConfigItem(
+		keyName = "rightClickBankEquip",
+		name = "Disable left click bank equipment",
+		description = "Configures whether the bank equipment button will bank your equipment on left click",
+		position = 1
+	)
+	default boolean rightClickBankEquip()
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: These should be an enum

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Accidental Typo, sorry

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Maybe this plugin actually do something when enabled? setting right click equip as default to true would make sense (and plugin disabled by default)

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Yeah that's a good idea

**Comment by Hydrox6** [comment]
Currently broken due to Tag-invetory and Tag-equipment options

**Comment by Hydrox6** [comment]
Closing until I can figure out if this is possible to fix

**Comment by Hydrox6** [comment]
Thanks to Adam, this PR now works again!

**Comment by Adam-** [comment]
This is #2867. Do we need a new plugin for this? We already have "banktags" and "bankvalue". Maybe the latter could be merged into this, banktags is sufficiently complex I think to justify being its own thing?

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java [hunk 8]
```diff
@@ -64,4 +64,37 @@ default boolean showExact()
...
+
+	@ConfigItem(
+		keyName = "rightClickBankEquip",
+		name = "Disable left click bank equipment",
+		description = "Configures whether the bank equipment button will bank your equipment on left click",
+		position = 5
+	)
+	default boolean rightClickBankEquip()
+	{
+		return true;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: false

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java [hunk 9]
```diff
@@ -23,14 +23,14 @@
...
-package net.runelite.client.plugins.bankvalue;
+package net.runelite.client.plugins.bank;
 
 import net.runelite.client.config.Config;
 import net.runelite.client.config.ConfigGroup;
 import net.runelite.client.config.ConfigItem;
 
-@ConfigGroup("bankvalue")
-public interface BankValueConfig extends Config
+@ConfigGroup("bank")
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: Won't this cause existing configs to be forgotten? Seems like we should migrate those.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankConfig.java]: I dont think it matters at all

**Pull request merged**